### PR TITLE
Configuration update for Arbitrum

### PR DESCRIPTION
**Pull request by arsen3d**
- [X ] **Logic** - thorough check (from everybody doing review)

### Summary
Deployment of Devnet with Arbitrum Nitro as standalone L2.
* All LP contracts are deployed.
* All services accounts are funded
* Output of configuration is visible in the log

### Task/Issue reference
https://github.com/Lilypad-Tech/web3/issues/3

### How to test this code? (optional)
To redeploy the chain run this workflow: 
https://github.com/Lilypad-Tech/lilypad/actions/workflows/devnet_deploy_chain.yml

This will reset the current deployment.

### CODE REVIEW

File: docker/chain/Dockerfile [hunk 1]
```diff
@@ -1,63 +1,72 @@
...
+
+RUN echo "tail -f /dev/null" >> run
+
+RUN chmod +x cloudflared-linux-amd64
 RUN chmod +x run
 
-CMD ["/bin/bash", "./run"]
+ENTRYPOINT ["bash", "./run"]  
+
+#example use:
```
Comment (AquiGorka) [inline docker/chain/Dockerfile]: Can you please add two commands to `stack`? One for building and one for running for local purposes.

Comment (AquiGorka) [reply docker/chain/Dockerfile]: And can you also add the `expose_via` mecahanism? This way, we can test the containerized version locally without having to setup cloudflare tunnels.

File: hardhat/DEPLOYMENT.MD [hunk 2]
```diff
@@ -0,0 +1,172 @@
+## Prerequisite
```
Comment (AquiGorka) [inline hardhat/DEPLOYMENT.MD]: This change should not be here

File: hardhat/hardhat.config.ts [hunk 3]
```diff
@@ -34,6 +34,32 @@ const config: HardhatUserConfig = {
       url: `https://sepolia.infura.io/v3/${INFURA_KEY}`,
       accounts: [getAccount('admin').privateKey],
     },
+    arbitrumOne: {
```
Comment (AquiGorka) [inline hardhat/hardhat.config.ts]: These changes should not be in this PR

File: hardhat/scripts/print-l2-config.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import {
```
Comment (AquiGorka) [inline hardhat/scripts/print-l2-config.ts]: This file should not be in this PR

File: hardhat/utils/web3.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ import {
 */
 
 // how much ether to send to each account
-export const DEFAULT_ETHER_PER_ACCOUNT = ethers.parseEther('1000')
+export const DEFAULT_ETHER_PER_ACCOUNT = ethers.parseEther('10')
```
Comment (AquiGorka) [inline hardhat/utils/web3.ts]: This change comes from another PR, it should not be here

File: pkg/options/configs/devnet.toml [hunk 6]
```diff
@@ -4,7 +4,7 @@ mediator = ["0x90F79bf6EB2c4f870365E785982E1f101E93b906"]
 
 [web3]
 rpc_url = "wss://devnet-chain-ws.lilypad.tech"
-chain_id = 421614
+chain_id = 412346
```
Comment (AquiGorka) [inline pkg/options/configs/devnet.toml]: Wait, is this correct? cc @bgins

File: stack [hunk 7]
```diff
@@ -354,4 +354,37 @@ function run() {
   doppler run --preserve-env -p run -c dev -- go run . run "$@"
 }
 
+
+function dev_deploy() {
```
Comment (AquiGorka) [inline stack]: These should not be here.

**Review by AquiGorka — COMMENTED** [review]
@arsen3d we need to review your workflow for stacking changes and committing changes when you have more than one PR open.

File: docker/chain/Dockerfile [hunk 8]
```diff
@@ -1,63 +1,72 @@
...
+RUN apk update
+RUN apk add bash
+RUN apk add nodejs
+RUN apk add npm
+
+WORKDIR /l2chain
+COPY . .
 RUN touch run
-RUN echo "#!/bin/bash" >> run
+RUN echo "dockerd-entrypoint.sh & " >> run
```
Comment (AquiGorka) [inline docker/chain/Dockerfile:17]: Let's add a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) to run `!#/bin/bash`

File: docker/chain/Dockerfile [hunk 9]
```diff
@@ -1,63 +1,72 @@
...
+RUN echo "docker pull offchainlabs/nitro-node:v2.3.4-b4cc111 " >> run
 
-FROM base AS expose-cloudflare
-RUN curl -L --output cloudflared.deb https://github.com/cloudflare/cloudflared/releases/latest/download/cloudflared-linux-${arch}.deb
-RUN dpkg -i cloudflared.deb
-RUN echo "cloudflared tunnel --metrics 0.0.0.0:11111 run --token $cloudflare_token_http --url http://localhost:8545 &" >> run
-RUN echo "cloudflared tunnel --metrics 0.0.0.0:11112 run --token $cloudflare_token_ws --url http://localhost:8546 &" >> run
+RUN echo "docker run -d --name devnet -p 0.0.0.0:8547:8547 -p 0.0.0.0:8548:8548 offchainlabs/nitro-node:v2.3.4-b4cc111  "\
+"   --init.dev-init " \
+"   --init.dev-init-address 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 "\
```
Comment (AquiGorka) [inline docker/chain/Dockerfile]: Let's pass this in as a `build-arg` (it was already in place before)

File: docker/chain/Dockerfile (hunk 8 above)
Comment (AquiGorka) [inline docker/chain/Dockerfile:17]: Is this a script that should be around?

Comment (arsen3d) [reply docker/chain/Dockerfile:17]: This is what is being called by nitro container.

File: docker/chain/Dockerfile (hunk 9 above)
Comment (arsen3d) [reply docker/chain/Dockerfile]: will parameterize admin address

**Review by AquiGorka — APPROVED** [review]
🎉 🎉 🍸 (margarita!)

Thanks @arsen3d !

File: hardhat/hardhat.config.ts [hunk 10]
```diff
@@ -34,7 +34,11 @@ const config: HardhatUserConfig = {
       chainId: CHAIN_ID,
       accounts: [getAccount('admin').privateKey],
     },
-
+    devnet: {
+      url: 'http://0.0.0.0:8547',
```
Comment (AquiGorka) [inline hardhat/hardhat.config.ts:38]: @arsen3d are we sure about this value? cc @bgins

**Pull request merged**